common/djangoapps/entitlements/tests/test_models.py

```diff
@@ -90,3 +102,30 @@ def test_get_days_until_expiration(self):
         # method will have had at least some time pass between object creation in setUp and this method execution,
         # or the exact same as the original expiration_period_days if somehow no time has passed
         assert entitlement.get_days_until_expiration() <= entitlement.policy.expiration_period_days
+
+    def test_expired_at_datetime(self):
+        """
+        Tests the using the getter method properly updates the expired_at field for an entitlement
+        """
+
+        # Verify a brand new entitlement isn't expired
+        entitlement = CourseEntitlementFactory.create()
+        assert entitlement.expired_at_datetime is None
+
+        # Verify an entitlement from two years ago is expired
+        past_datetime = datetime.utcnow().replace(tzinfo=pytz.UTC) - timedelta(days=365 * 2)
+        entitlement.created = past_datetime
+        entitlement.save()
+        assert entitlement.expired_at_datetime
+
+        # Verify that a brand new entitlement that has been redeemed is not expired
+        entitlement = CourseEntitlementFactory.create(enrollment_course_run=self.enrollment)
+        assert entitlement.enrollment_course_run
+        assert entitlement.expired_at_datetime is None
+
+        # Verify that an entitlement that has been redeemed but not within 14 days is expired
+        past_datetime = datetime.utcnow().replace(tzinfo=pytz.UTC) - timedelta(days=15)
+        entitlement.created = past_datetime
+        entitlement.save()
+        assert entitlement.enrollment_course_run
+        assert entitlement.expired_at_datetime
```

Comment:
The assertions are good, but we should make sure it updates the DB, not just that it returns a value.